static/js/chartHandler.js

```diff
@@ -1,45 +1,175 @@
+      ranges[2].default_max = not_loaded ? urlParams.get('ra') : $('#input-text-ra')[0].value;
+  }
+
+  //Limits max value of x to lengths current value, since the x < length
+  if (x_value === 4) {
+    ranges[4].max = not_loaded ? urlParams.get('length') : $('#input-text-length')[0].value;
+    ranges[4].default_max = not_loaded ? urlParams.get('length') : $('#input-text-length')[0].value;
+  }
+
+  //Jquery slider, for information can be found here:  https://jqueryui.com/slider/#range
+  $('#slider-range').slider({
+    range: true,
+    min: ranges[x_value].min,
+    max: ranges[x_value].max,
+    step: ranges[x_value].step,
+    values: [ ranges[x_value].default_min, ranges[x_value].default_max ],
+    slide: (event, ui) => {
+      if (event.originalEvent) {
+        $('#x-value-range').val(ui.values[0] + ' - ' + ui.values[1]);
+      }
+    },
+  });
+  $('#x-value-range').val(
+    $('#slider-range').slider('values', 0) +' - ' + $('#slider-range').slider('values', 1)
+  );
+  let slider_min = $('#slider-range').slider('values', 0);
+  let slider_max = $('#slider-range').slider('values', 1);
+  $('#step-input')[0].max = slider_max - slider_min - 0.01;
+};
+
+
+//Creates chart
 const chartHandler = () => {
   let values = inputHandler();
-  let xGraph = $('#x-values-input :selected').text().toLowerCase();
-  let yGraph = $('#y-values-input :selected').text().toLowerCase();
-  let xStart = $('#slider-range').slider("values")[0];
-  let xEnd = $('#slider-range').slider("values")[1];
-  let xStep = $('#step-input')[0].value
 
-  values.blanks = values.blanks.filter(i => i !== xGraph && i !== yGraph);
+  //User inputs from drop downs and sliders needed for the graph
+  let x_graph = $('#x-values-input :selected').text().toLowerCase();
+  let y_graph = $('#y-values-input :selected').text().toLowerCase();
+  let x_start = $('#slider-range').slider("values")[0];
+  let x_end = $('#slider-range').slider("values")[1];
+  let x_step = $('#step-input')[0].value;
 
-  if(xGraph === 'Select X Value' || yGraph === 'Select Y Value'){
+  values.blanks = values.blanks.filter(i => i !== x_graph && i !== y_graph);
+
+  //-------------------------Error handling-----------------------------
+ 
+  //If defaults
+  if(x_graph === 'Select X Value' || y_graph === 'Select Y Value'){
     flashHandler('PLEASE SELECT X AND Y VALUES', 'same-selections-flash-err');
     return;
   }
 
+  //Needing correct inputs from the text-inputs
   if(values.blanks.length >= 1) {
-    flashHandler(`ALL VALUES OTHER THAN ${xGraph} AND ${yGraph} NEED TO BE FILLED `, 'missing-input-flash-err');
+    flashHandler(`ALL VALUES OTHER THAN ${x_graph} AND ${y_graph} NEED TO BE FILLED `, 'missing-input-flash-err');
     return;
   }
-  if(xGraph === 'x' && yGraph === 'length'){
+
+  //x vs length is unable to graph
+  if(x_graph === 'x' && y_graph === 'length'){
     flashHandler(`UNABLE TO GRAPH X VS LENGTH, PLEASE CHOOSE OTHER VALUES `, 'x-vs-length-input-flash-err');
```

Comment:
Do you think it would be worthwhile to have these as constant strings?
Like at the top of the file have something like we did for the Solenoid parameters:

```javascript
const ErrorMessages = {
   DEFAULT: 'PLEASE SELECT X AND Y VALUES'
}